bluepyopt/ephys/create_acc.py

```diff
@@ -0,0 +1,832 @@
+                    if cls._is_global_property(loc, param):
+                        global_properties.append(param)
+                    else:
+                        local_properties.append(param)
+                local_mechs[mech] = local_properties
+            else:
+                local_mechs[mech] = params
+
+        return local_mechs, {None: global_properties}
+
+    @staticmethod
+    def _format_params_and_group_by_mech(params, channels):
+        """Group list of parameters by mechanism and turn them to Arbor format
+
+        Args:
+            params (): List of parameters in Neuron format
+            channels (): List of co-located NMODL mechanisms
+
+        Returns:
+            Mapping of Arbor mechanism name to list of parameters in Arbor
+            format
+        """
+        mech_params = [Nrn2ArbParamFormatter.format(
+                       param, channels) for param in params]
+        mechs = {mech: [] for mech, _ in mech_params}
+        for mech in channels:
+            if mech not in mechs:
+                mechs[mech] = []
+        for mech, param in mech_params:
+            mechs[mech].append(param)
+        return mechs
+
+    @classmethod
+    def process_global(cls, params):
+        """Group global params by mechanism, convert them to Arbor format
+        Args:
+            params (): List of global parameters in Neuron format
+
+        Returns:
+            A mapping of mechanism to parameters. The mechanism parameters are
+            in Arbor format (mechanism name is None for non-mechanism
+            parameters).
+        """
+        return cls._format_params_and_group_by_mech(
+            [Location(name=name, value=value)
+             for name, value in params.items()],
+            []  # no default mechanisms
+        )
+
+    @classmethod
+    def process_local(cls, params, channels):
+        """Group local params by mechanism, convert them to Arbor format
+
+        Args:
+            params (): List of Arbor label/local parameters pairs in Neuron
+            format
+            channels (): Mapping of Arbor label to co-located NMODL mechanisms
+
+        Returns:
+            In the first component, a two-level mapping of Arbor label to
+            mechanism to parameters. The mechanism parameters are in Arbor
+            format (mechanism name is None for non-mechanism parameters).
+            In the second component, the global properties found are returned.
+        """
+        local_mechs = dict()
+        global_properties = dict()
+        for loc, loc_params in params:
+            mechs = cls._format_params_and_group_by_mech(
+                loc_params, channels[loc])
+
+            # move Arbor global properties to global_params
+            mechs, global_props = cls._separate_global_properties(loc, mechs)
+            for mech, props in global_props.items():
+                global_properties[mech] = \
+                    global_properties.get(mech, []) + props
```

Comment:
This relates to the discussion above again. Indeed, `mech` does carry information, because `None` refers to non-mechanism parameters. The loop should be irrelevant performance-wise. For the rest, see above.

Reply:
I replied above. The function return value can stay if `None` really is informative, but the for loop here should be removed.

Reply:
See above - should be resolved.

Reply:
Thanks